beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -179,128 +158,139 @@ impl<E: EthSpec> PendingComponents<E> {
-        spec: &Arc<ChainSpec>,
-    ) -> Option<AvailableBlockData<E>> {
-        let Some(block) = self.get_cached_block() else {
-            // Block not available yet
-            return None;
-        };
-
-        let num_expected_blobs = block.num_blobs_expected();
-        if num_expected_blobs == 0 {
-            return Some(AvailableBlockData::NoData);
-        }
-
-        if spec.is_peer_das_enabled_for_epoch(block.epoch()) {
-            if self.verified_data_columns.len() >= custody_column_count {
-                // Block is post-peerdas, and we got enough columns
-
-                // TODO(das) get only the columns that we need, filter the rest
-
-                let data_columns = self
-                    .verified_data_columns
-                    .iter()
-                    .map(|d| d.clone().into_inner())
-                    .collect::<Vec<_>>();
-                return Some(AvailableBlockData::DataColumns(data_columns));
-            }
-
-            if let Some(data_columns_recv) = self.data_column_recv.take() {
-                // The data_columns_recv is an infallible promise that we will receive all expected
-                // columns, so we consider the block available
-                return Some(AvailableBlockData::DataColumnsRecv(data_columns_recv));
-            }
-
-            None
-        } else {
-            // Before PeerDAS, blobs
-            let num_received_blobs = self.verified_blobs.iter().flatten().count();
-            if num_received_blobs >= num_expected_blobs {
-                // TODO(das): Should do something special if `num_received_blobs > num_expected_blobs`?
-                let blobs_vec = self
-                    .verified_blobs
-                    .iter()
-                    .flatten()
-                    .map(|blob| blob.clone().to_blob())
-                    .collect::<Vec<_>>();
-                let max_blobs = spec.max_blobs_per_block(block.epoch());
-                let blobs = RuntimeVariableList::new(blobs_vec, max_blobs as usize)
-                    .expect("num_expect_blobs is less than max_blobs");
-                Some(AvailableBlockData::Blobs(blobs))
-            } else {
-                // Not enough blobs received yet
-                None
-            }
-        }
-    }
-
-    /// Returns an empty `PendingComponents` object with the given block root.
-    pub fn empty(block_root: Hash256, max_len: usize) -> Self {
-        Self {
-            block_root,
-            verified_blobs: RuntimeFixedVector::new(vec![None; max_len]),
-            verified_data_columns: vec![],
-            executed_block: None,
-            reconstruction_started: false,
-            data_column_recv: None,
-        }
-    }
-
     /// Verifies an `SignedBeaconBlock` against a set of KZG verified blobs.
     /// This does not check whether a block *should* have blobs, these checks should have been
     /// completed when producing the `AvailabilityPendingBlock`.
```

Comment:
Do you mean `AvailabilityPendingExecutedBlock` here?
```suggestion
    /// completed when producing the `AvailabilityPendingExecutedBlock`.
```

Reply:
Just realised this is existing comments that is outdated.